site/en/docs/extensions/reference/storage/index.md

```diff
@@ -2,26 +2,65 @@
-- Your extension's content scripts can directly access user data without the need for a background
-  page.
-- A user's extension settings can be persisted even when using [split incognito
-  behavior][incognito].
-- It's asynchronous with bulk read and write operations, and therefore faster than the blocking and
-  serial `localStorage API`.
-- User data can be stored as objects (the `localStorage API` stores data in strings).
-- Enterprise policies configured by the administrator for the extension can be read (using
-  `storage.managed` with a [schema][api-storage]).
+- Stores JSON serializable values as object properties.
+- Storage is asynchronous with bulk read and write operations.
+- Storage data persists when the user clears the cache and browsing history.
+- Available in the service worker, content scripts, and other extension contexts.
+- A user's extension settings persist even when using [split incognito][incognito].
+- Offers an exclusive read-only [managed storage area][manifest-storage] for enterprise policies.
 
-## Manifest
+{% Details %}
+{% DetailsSummary %}
+💡 Can extensions use Window.localStorage?
+{% endDetailsSummary %}
 
-You must declare the "storage" permission in the [extension manifest][doc-manifest] to use the storage
-API. For example:
+Even though extensions can access [Window.localStorage][mdn-localstorage] in some contexts (popup and other HTML pages), it is not recommended for the following reasons:
+
+- The service worker cannot access localStorage API.
+- Content scripts share the localStorage of the host page.
+- The browser clears any data stored by the extension in localStorage when the user clear their browsing history and data for privacy reasons.
+
+{% endDetails %}
+
+### Storage areas
+
+The Storage API is divided into the following four buckets ("storage areas"): 
+
+- [**storage.local**][prop-local]
+  - Data is stored locally, which is cleared when the extension is removed.
+  - Quota limitation is approx 5 MB, but can be increased by requesting the `"unlimitedStorage"` permission.
+  - Consider using it to store larger amounts of data.
+
+- [**storage.sync**][prop-sync]
+  - If syncing is enabled, the data is synced to any Chrome browser that the user is logged into. If disabled, it behaves like `storage.local`.
+  - When the browser is offline, Chrome stores the data locally and resumes syncing when it's back online.
+  - Quota limitation is 100 KB approx, 8 KB per item
+  - Consider using it to preserve user settings across synced browsers. 
+
+{% Aside 'warning' %}
+
+Local and sync storage areas should not store confidential user data because they are not encrypted. We recommend using the `session` storage area, which is not saved on disk for sensitive data.
+
+{% endAside %}
+
+- [**storage.session**][prop-session]
+  - Holds small amounts of data in memory for the duration of a browser session.
+  - Not available in content scripts.
```

Comment:
This sounded odd to me, so I just tested it and my content script was able to access data in both the sync and local storage areas.

```suggestion
```

Reply:
@dotproto Did you test the [storage.session](https://developer.chrome.com/docs/extensions/reference/storage/#property-session) in content scripts?

Reply:
> - Not available in content scripts.

@dotproto Makes sense. How do we want to proceed?

A. Change list item to: Not exposed to content scripts by default, but behavior can be changed by setting [chrome.storage.session.setAccessLevel()](https://developer.chrome.com/docs/extensions/reference/storage/#method-StorageArea-setAccessLevel).
B. Add the example you included.
C. Remove "Not available in content scripts." list item.

Reply:
Let's run with option A for the moment.